task2/list.c

```diff
@@ -0,0 +1,151 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "list.h"
+
+typedef struct Node
+{
+    Value value;
+    struct Node *next;
+} Node;
+
+typedef struct List
+{
+    Node *head;
+} List;
+
+Error listCreate(List **list)
+{
+    *list = calloc(1, sizeof(List));
+    if (*list == NULL)
+    {
+        return MemoryAllocationError;
+    }
+    return OK;
+}
+
+Error addValue(List *list, Value value)
+{
+    if (list == NULL)
+    {
+        return PointerIsNull;
+    }
+    if (list->head == NULL)
+    {
+        list->head = calloc(1, sizeof(Node));
+        list->head->value = value;
+        return OK;
+    }
+
+    Node *newNode = calloc(1, sizeof(Node));
+    if (newNode == NULL)
+    {
+        return PointerIsNull;
+    }
+    newNode->value = value;
+
+    Node *currentNode = list->head;
+    Node *previousNode = NULL;
+    while (currentNode->value < newNode->value && currentNode->next != NULL)
+    {
+        previousNode = currentNode;
+        currentNode = currentNode->next;
+    }
+    if (currentNode == list->head && currentNode->value < newNode->value)
+    {
+        currentNode->next = newNode;
+    }
+    else if (previousNode == NULL) // в начало
+    {
+        newNode->next = currentNode;
+        list->head = newNode;
+    }
+    else if (currentNode->value > newNode->value) // в середину
+    {
+        previousNode->next = newNode;
+        newNode->next = currentNode;
+    }
+    else // в конец
+    {
+        currentNode->next = newNode;
+        newNode->next = NULL;
+    }
+    return OK;
+}
+
+Error removeValue(List *list, Index index, Value *deletedValue)
```

Comment:
```suggestion
Error removeByIndex(List *list, Index index, Value *deletedValue)
```
Название в заблуждение вводит. Кажется, что нужно передать значение для удаления, судя по названию функции.